### DISCUSSION TITLE
migrate from localhost to shared hosting

### DISCUSSION THREAD

**Discussion by veeyka** [peers/peerjs#1220] [Q&A] [answered]
hi . peer server and peerjs client is working well on my localhost . i brought it on my shared hosting . now i have problem connecting to peer server after uploading it to my host . its 2 days im chaning options on both server and client side and got nothing .
i made a subdomain in my host for peer server  for runing express . i think my problem is that host dont start my server with port 9000 which i have set .

its my server address . when you click it its says peer server is working as it have to be .
https://node2.bonyadonline.com/peer-server

--------- my server code is :

```ts
const express = require('express');
const fs = require("fs");
const app = express();
const { ExpressPeerServer } = require('peer');
const cors = require("cors");
app.use(cors());

const server2 = app.listen(9000);

const peerServer = ExpressPeerServer(server2, {
  debug: true,
  allow_discovery: true,
  ssl: {
    key: fs.readFileSync('./keyname.key'),
    cert: fs.readFileSync('./cername.cer')
    }
});

app.use("/peer-server",peerServer);
```

---------- and my client code is : 

```ts
const peer = new Peer(undefined,{
  secure:true,
  key:"peerjs",
  host: "node2.bonyadonline.com",
  port: "9000",
  path: '/peer-server',
});
```

could you please help me . thanks

**Comment by jonasgloning** [comment, marked as answer]
It looks like you run nginx in front of the PeerJS server. There is probably a misconfiguration there, as the server code looks fine and it did work locally.

Plus you need to remove the following line from the client config:
```ts
port: "9000",
```

**Reply by veeyka** [reply]
yes dude . you right . i did some steps  .
1 - i went to my hosting in ngnix configuration and unchecked proxy mod . 
2 - in client options remove port and key box . like this 

`const peer = new Peer(undefined,{
  secure:true,
  host: "node2.bonyadonline.com",
  path: '/peer-server',
});`

**Reply by umarali07** [reply]
@veeyka have you install on cpanel? if yes what is version of node.js is on your cpanel
